lang/src/accounts/program.rs

```diff
@@ -75,13 +71,7 @@ impl<'a, T: Id + AccountDeserialize + Clone> Program<'a, T> {
             None
         };
 
-        // Programs have no data so use an empty slice.
-        let mut empty = &[][..];
-        Ok(Program::new(
-            info.clone(),
-            T::try_deserialize(&mut empty)?,
```

Comment:
We should remove this trait bound and the associated implementations as well.

Reply:
the `Deserialize` trait bound for `T`?

Reply:
yea makes sense

Reply:
the `dex` struct also implements `AccountSerialize`. Was that intentional?

Reply:
Sounds like a mistake. Can remove that too.

Reply:
done